js/tests.js

```diff
@@ -52,4 +52,47 @@ describe("get auto-complete context", function() {
         expect(parser.autoCompleteContext('Hi @(CONCAT("@con')).toBeNull();
         expect(parser.autoCompleteContext('Hi @("!" & "@con')).toBeNull();
     });
+
+    it("ignore parenthesis triggering functions completions for variables", function() {
+        expect(parser.autoCompleteContext("Hi @(contact.age")).toBe('contact.age');
+    });
+
+    it('ignore the parenthesis triggering functions completions for functions', function() {
+        expect(parser.autoCompleteContext("Hi @(SUM")).toBe('SUM');
+    });
+
+    it("matches the function without parameters", function() {
+        expect(parser.autoCompleteContext("Hi @(SUM(")).toBe('SUM');
+    });
+
+    it("matches the function missing balanced parentheses", function() {
+        expect(parser.autoCompleteContext("Hi @(SUM(dads, ABS(number))")).toBe('SUM');
+        expect(parser.autoCompleteContext("Hi @(SUM(dads, ABS(number)")).toBe('ABS');
+    });
+
+    it("ignores trailing spaces in function parameters", function () {
+        expect(parser.autoCompleteContext("Hi @(SUM( ")).toBe('SUM');
+        expect(parser.autoCompleteContext("Hi @(SUM(   ")).toBe('SUM');
+        expect(parser.autoCompleteContext("Hi @(SUM(dads, ABS(number))  ")).toBe('SUM');
+        expect(parser.autoCompleteContext("Hi @(SUM(dads, ABS(number)  ")).toBe('ABS');
```

Comment:
Again ABS call is closed here so it should go back to showing summary box for SUM